docs/source/developers/java/building.rst

```diff
@@ -0,0 +1,214 @@
+    Homebrew Bundle complete! 25 Brewfile dependencies now installed.
+    $ export JAVA_HOME=<absolute path to your java home>
+    $ mkdir -p java-dist java-native-cpp
+    $ cd java-native-cpp
+    $ cmake \
+        -DARROW_BOOST_USE_SHARED=OFF \
+        -DARROW_BROTLI_USE_SHARED=OFF \
+        -DARROW_BZ2_USE_SHARED=OFF \
+        -DARROW_GFLAGS_USE_SHARED=OFF \
+        -DARROW_GRPC_USE_SHARED=OFF \
+        -DARROW_LZ4_USE_SHARED=OFF \
+        -DARROW_OPENSSL_USE_SHARED=OFF \
+        -DARROW_PROTOBUF_USE_SHARED=OFF \
+        -DARROW_SNAPPY_USE_SHARED=OFF \
+        -DARROW_THRIFT_USE_SHARED=OFF \
+        -DARROW_UTF8PROC_USE_SHARED=OFF \
+        -DARROW_ZSTD_USE_SHARED=OFF \
+        -DARROW_JNI=ON \
+        -DARROW_PARQUET=ON \
+        -DARROW_FILESYSTEM=ON \
+        -DARROW_DATASET=ON \
+        -DARROW_GANDIVA_JAVA=ON \
+        -DARROW_GANDIVA_STATIC_LIBSTDCPP=ON \
+        -DARROW_GANDIVA=ON \
+        -DARROW_ORC=ON \
+        -DARROW_PLASMA_JAVA_CLIENT=ON \
+        -DARROW_PLASMA=ON \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_LIBDIR=lib \
+        -DCMAKE_INSTALL_PREFIX=../java-dist \
+        -DCMAKE_UNITY_BUILD=ON \
+        -Dre2_SOURCE=BUNDLED \
+        -DBoost_SOURCE=BUNDLED \
+        -Dutf8proc_SOURCE=BUNDLED \
+        -DSnappy_SOURCE=BUNDLED \
+        -DORC_SOURCE=BUNDLED \
+        -DZLIB_SOURCE=BUNDLED \
+        ../cpp
+    $ cmake --build . --target install
+    $ ls -latr  ../java-dist/lib
+    |__ libarrow_dataset_jni.dylib
+    |__ libarrow_orc_jni.dylib
+    |__ libgandiva_jni.dylib
+
+Building Arrow JNI Modules
+--------------------------
+
+To compile the JNI bindings, use the ``arrow-c-data`` Maven profile:
+
+.. code-block::
+
+    $ cd arrow/java
+    $ mvn -Darrow.c.jni.dist.dir=../java-dist/lib -Parrow-c-data clean install
+
+To compile the JNI bindings for ORC / Gandiva / Dataset, use the ``arrow-jni`` Maven profile:
+
+.. code-block::
+
+    $ cd arrow/java
+    $ mvn -Darrow.cpp.build.dir=../java-dist/lib -Parrow-jni clean install
+
+IDE Configuration
+=================
+
+IntelliJ IDE
+------------
+
+IntelliJ can import Maven projects. To start working on Arrow in IntelliJ, just open the `java/`
+subdirectory of the Arrow repository.
+
+.. figure:: img/java_welcome.png
+   :scale: 50 %
+   :alt: A newly opened Arrow project
+
+An option to understand the arrow java code is running the java unit test in debugging mode to see
+objects initialized parameters and values assigned.
+
+Let's debug `TestArrowBuf.java`_ to see `ArrowBuf` object initialization. Go to the java class `TestArrowBuf.java`
+and method `testSetBytesBigEndian`, add breakpoints needed, right click on the method name and select option
+mention `Debug 'testSetBytesBigEndian()'` and inspect te variable when debug mode arrive to `ArrowBuf` variable.
+
+.. figure:: img/java_debug_mode.png
+   :scale: 50 %
+   :alt: An ArrowBuf variable initialized with parameters and values assigned.
```

Comment:
Again: I think this is all very basic stuff for an IntelliJ user. The reason why I brought up IDEs in the first place was because I recalled that IntelliJ needed some settings in order to build Arrow (due to using error-prone). If that's not the case anymore, then all we need to do is tell people to open the Java subdirectory, and I don't think we need to add screenshots or anything else. We shouldn't add more text for the sake of adding it.

Reply:
Oh, ok, I get it. Let me test error-prone and add if is something is needed, if not only add  a text mention open the project for modules needed

Reply:
The other thing is that I recall builds don't work for me as the generated code is not indexed by IntelliJ by default. But that may be down to doing a build instead of a Maven build.

Reply:
Updated